packages/chatbot-server-mongodb-public/src/config.ts

```diff
@@ -225,42 +228,69 @@ const llmAsAJudgeConfig = {
   },
 };
 
+const segmentConfig = SEGMENT_WRITE_KEY
+  ? {
+      writeKey: SEGMENT_WRITE_KEY,
+    }
+  : undefined;
+
 export const config: AppConfig = {
   conversationsRouterConfig: {
     llm,
     middleware: [
       blockGetRequests,
       requireValidIpAddress(),
       requireRequestOrigin(),
+      useSegmentIds(),
       cookieParser(),
     ],
     createConversationCustomData: !isProduction
       ? createConversationCustomDataWithAuthUser
       : undefined,
+    addMessageToConversationCustomData: async (req, res) => {
+      const defaultCustomData = await defaultAddMessageToConversationCustomData(
+        req,
+        res
+      );
+      return {
+        ...defaultCustomData,
+        segmentUserId: res.locals.customData.segmentUserId ?? "NOPE",
+        segmentAnonymousId: res.locals.customData.segmentAnonymousId ?? "NOPE",
```

Comment:
can you do `null` rather than `'NOPE'`? or does 'NOPE' mean something to segment

Reply:
Oops did that for testing purposes and used a silly word so that I'd notice it later. Good catch!